packages/asl-search/bin/indexer

```diff
@@ -1,75 +1,199 @@
 #!/usr/bin/env node
 
 try {
-  // eslint-disable-next-line implicit-dependencies/no-implicit
   require('dotenv').config();
 } catch (e) {}
 
 const { get } = require('lodash');
 const minimist = require('minimist');
-const { green } = require('chalk');
+const { green, yellow, red } = require('chalk');
 const winston = require('winston');
 const settings = require('../config');
 const db = require('../lib/db');
 const taskflowDb = require('../lib/db/taskflow');
 const { createESClient } = require('../lib/elasticsearch');
-
 const { indexers } = require('../lib/indexers');
 
-const start = process.hrtime();
 const args = minimist(process.argv.slice(2));
 const model = args._[0] || 'all';
 
 const logger = winston.createLogger({
   level: 'debug',
-  transports: [ new winston.transports.Console({ level: process.env.LOG_LEVEL || 'info' }) ]
+  transports: [new winston.transports.Console({ level: process.env.LOG_LEVEL || 'info' })]
 });
 
-Promise.resolve()
-  .then(() => createESClient(settings.es))
-  .then(esClient => {
-    const index = key => {
-      console.log(`Indexing ${green(key)}`);
+class IndexManager {
+  constructor(esClient, options = {}) {
+    this.esClient = esClient;
+    this.options = options;
+    this.stats = {
+      startTime: Date.now(),
+      processed: 0,
+      errors: 0
+    };
+
+    // Define large indexes that need special handling
+    this.largeIndexes = ['projects', 'projects-content', 'tasks'];
+  }
+
+  async indexAll() {
+    console.log(yellow('Starting sequential indexing for all models...'));
+
+    const indexKeys = Object.keys(indexers);
+    const results = [];
+
+    // Process ONE index at a time (sequential) for large datasets
+    const BATCH_SIZE = 1;
+
+    for (let i = 0; i < indexKeys.length; i += BATCH_SIZE) {
+      const batch = indexKeys.slice(i, i + BATCH_SIZE);
+      const currentKey = batch[0];
+
+      console.log(yellow(`Processing: ${currentKey}`));
+
+      const batchResults = await Promise.allSettled(
+        batch.map(key => this.indexSingle(key))
+      );
+
+      results.push(...batchResults);
+
+      // Longer pause between indexes, especially after large ones
+      if (i + BATCH_SIZE < indexKeys.length) {
+        const delayTime = this.largeIndexes.includes(currentKey) ? 10000 : 2000;
+        console.log(yellow(`Cooling down for ${delayTime/1000} seconds...`));
+        await this.delay(delayTime);
+      }
+    }
```

Comment:
If batch size is always one, we don't need the complexity of batches and this would be easier to read and maintain as a foreach loop.